planner/core/logical_plan_builder.go

```diff
@@ -180,6 +180,33 @@ func (b *PlanBuilder) buildResultSetNode(node ast.ResultSetNode) (p LogicalPlan,
 	}
 }
 
+// pushDownConstExpr checks if the condition is from filter condition, if true, push it down to both
```

Comment:
What does `filter` here mean?

Reply:
To differentiate the where conditions from join conditions, I use `filter condition` here. I can change it to another name if it is misleading.

Reply:
So it should be the conditions exclude the join key conditions? If so, can `deriveLeft || deriveRight` match its meaning?

Reply:
`extractOnCondition` is called by `PredicatePushDown` and `atttachOnConds`, in `PredicatePushDown`, the input conditions are treated as filter conditions, and at least `deriveLeft` or `deriveRight` is true; in `attachOnConds`, the input conditions are join conditions, and `deriveLeft` and `deriveRight` are both false, so it should be correct?